android/cli/lib/process-drawables-task.js

```diff
@@ -0,0 +1,64 @@
+const path = require('path');
+const CopyResourcesTask = require('../../../cli/lib/tasks/copy-resources-task');
+
+const appc = require('node-appc');
+const i18n = appc.i18n(__dirname);
+const __ = i18n.__;
+
+const drawableDpiRegExp = /^(high|medium|low)$/;
+
+/**
+ * Task that copies Android drawables into the app.
+ */
+class ProcessDrawablesTask extends CopyResourcesTask {
+
+	/**
+	 * Constructs a new processing task.
+	 *
+	 * @param {Object} options Configuration object for this task.
+	 * @param {String} [options.name='process-drawables'] Name for the task.
+	 * @param {String} options.incrementalDirectory Path to a folder where incremental task data will be stored.
+	 * @param {Map<string,FileInfo>} options.files Map of input files to file info
+	 * @param {Object} [options.logger] The logger instance used by this task.
+	 * @param {Object} options.builder Builder instance.
+	 */
+	constructor(options) {
+		options.name = options.name || 'process-drawables';
+		const appMainResDir = options.builder.buildAppMainResDir;
+
+		// We have to modify the destination paths for each file. Is there a place we can move this to other than constructor?
+		// Because we want the expected final filename for whenever the incremental task stuff needs it
+		options.files.forEach((info, relPath) => {
+			const parts = relPath.split(path.sep);
+			const origFoldername = parts[1];
+			const foldername = drawableDpiRegExp.test(origFoldername) ? 'drawable-' + origFoldername[0] + 'dpi' : 'drawable-' + origFoldername.substring(4);
+
+			const destFilename = `${info.name}.${info.ext}`;
+
+			// We have a drawable image file. (Rename it if it contains invalid characters.)
+			const warningMessages = [];
+			if (parts.length > 3) {
+				warningMessages.push(__('- Files cannot be put into subdirectories.'));
+			}
+			let destFilteredFilename = `${info.name.toLowerCase().replace(/[^a-z0-9_]/g, '_')}.${info.ext}`;
```

Comment:
This breaks 9-patch image handling by turning "my.9.png" to "my_9.png". We need to keep the ".9.png" part. This is the only case where a 2nd period '.' in the file name is okay.

So perhaps do the following instead?
```javascript
let destFilteredFilename = info.name.toLowerCase();
let destFilteredExtension = info.ext;
if (destFilteredFilename.endsWith('.9') && (info.ext === 'png')) {
	destFilteredFilename = destFilteredFilename.substring(0, destFilteredFilename.length - 2);
	destFilteredExtension = '9.png';
}
destFilteredFilename = `${destFilteredFilename.replace(/[^a-z0-9_]/g, '_')}.${destFilteredExtension}`;
```

Reply:
Playing with the regex I was able to get this, which should acheive the same result:
```JS
let destFilteredFilename = `${info.name.toLowerCase().replace(/(?!\.9$)[^a-z0-9_]/g, '_')}.${info.ext}`;
```